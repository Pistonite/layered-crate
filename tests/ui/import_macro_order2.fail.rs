mod current {
    pub struct MyTypeInThisLayer;
    pub struct MyDepInThisLayer;
    pub mod crate_ {
        pub mod m {
            pub fn a() {}
            pub fn b() {}
        }
        pub mod my_other_dep {
            pub struct MyOtherType;
        }
    }
}

mod test {
    #[layered_crate::import]
    use current::{
        super::my_other_dep::MyOtherType,
        self::MyTypeInThisLayer,
        super::m::{a, b},
        self::MyDepInThisLayer,
    };
}

fn main() {}
