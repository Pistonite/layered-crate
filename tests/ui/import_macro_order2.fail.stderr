error: Please group self (current layer) imports and other imports separately for readability.
       Move this import up to be together with other `super` imports
  --> tests/ui/import_macro_order2.fail.rs:20:9
   |
20 |         super::m::{a, b},
   |         ^^^^^
